### PR TITLE
build: add support for testing more of the compilers on Windows

### PR DESCRIPTION
**Pull request by compnerd**
This adds the ability to conditionally run the test suite for LLVM, LLD, LLDB, Clang, and Swift.  This will allow us to increase the test coverage in CI to avoid regressions.

### CODE REVIEW

**Comment by compnerd** [comment]
CC: @tristanlabelle

**Comment by compnerd** [comment]
@swift-ci please test

**Comment by compnerd** [comment]
CC: @shahmishal

File: utils/build.ps1 [hunk 1]
```diff
@@ -1801,7 +1819,17 @@ if ($Stage) {
...
-if ($Test -contains "swift") { Build-Compilers $HostArch -Test }
+if ($Test -ne $null -and (Compare-Object $Test @("clang", "lld", "lldb", "llvm", "swift") -PassThru -IncludeEqual -ExcludeDifferent) -ne $null) {
+  $Tests = @{
+    "-TestClang" = $Test -contains "clang";
+    "-TestLLD" = $Test -contains "lld";
+    "-TestLLDB" = $Test -contains "lldb";
+    "-TestLLVM" = $Test -contains "llvm";
+    "-TestSwift" = $Test -contains "swift";
+  }
+  Build-Compilers $HostArch @Tests
```
Comment (tristanlabelle) [inline utils/build.ps1:1830]: You could avoid splatting by using `-TestClang:($Test -contains "clang")` (`-Foo:$true` / `-Foo:$false` is valid syntax)

**Review by tristanlabelle — APPROVED** [review]

File: utils/build.ps1 (hunk 1 above)
Comment (compnerd) [reply utils/build.ps1:1830]: Yeah, that is certainly a viable option as well.  I find it difficult to tell which is better.  The collection as is does make it easier to tell at a glance what options are being setup.  I would defer to your expertise if one is better than the other.

Comment (compnerd) [reply utils/build.ps1:1830]: I'm going to merge this for now, but happy to revisit if you have suggestions here.

**Pull request merged**